src/GraphQL/Types/Scalars/ScalarGraphType.cs

```diff
@@ -41,21 +41,20 @@ public abstract class ScalarGraphType : GraphType
         /// server-side representation. Input coercion may not only return primitive values like
         /// String but rather complex ones when appropriate.
         /// <br/><br/>
-        /// This method must handle a value of <see cref="NullValue"/>.
+        /// This method must handle a value of <see cref="GraphQLNullValue"/>.
+        /// <br/><br/>
+        /// This method SHOULD be overridden by descendants.
         /// </summary>
-        /// <param name="value">AST value node. Must not be <see langword="null"/>, but may be <see cref="NullValue"/>.</param>
+        /// <param name="value">AST value node. Must not be <see langword="null"/>, but may be <see cref="GraphQLNullValue"/>.</param>
         /// <returns>Internal scalar representation. Returning <see langword="null"/> is valid.</returns>
-        public virtual object? ParseLiteral(IValue value) => value switch
+        public virtual object? ParseLiteral(GraphQLValue value) => value switch
         {
-            BooleanValue b => ParseValue(b.Value.Boxed()),
-            IntValue i => ParseValue(i.Value),
-            LongValue l => ParseValue(l.Value),
-            BigIntValue bi => ParseValue(bi.Value),
-            FloatValue f => ParseValue(f.Value),
-            DecimalValue d => ParseValue(d.Value),
-            StringValue s => ParseValue(s.Value),
-            NullValue _ => ParseValue(null),
-            _ => ThrowLiteralConversionError(value)
+            GraphQLBooleanValue b => ParseValue(b.BoolValue.Boxed()),
+            GraphQLIntValue i => ParseValue((string)i.Value), //ISSUE:allocation
```

Comment:
Not a big deal since `ParseLiteral` default implementation SHOULD be overridden.